tokio/src/net/tcp/stream.rs

```diff
@@ -559,6 +561,14 @@ impl TcpStream {
             .try_io(Interest::READABLE, || (&*self.io).read(buf))
     }
 
+    /// Try to read data from the stream into the provided buffer, advancing the
+    /// buffer's internal cursor, returning how many bytes were read.
```

Comment:
Additionally the documentation could use an example. You can take inspiration from the existing `try_read` methods, but reading into a vector created using `Vec::with_capacity(1024)` to show how you can read into the unused capacity of a vector.